cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java

```diff
@@ -104,71 +101,46 @@ public PublishResource(@Named("config:se.simonsoft.cms.hostname") String hostnam
 	}
 	
 	private static final Logger logger = LoggerFactory.getLogger(PublishResource.class);
-	
-	@GET
-	@Path("release")
-	@Produces(MediaType.TEXT_HTML)
-	public String getReleaseForm(@QueryParam("item") CmsItemIdArg itemId) throws Exception {
-		
-		if (itemId == null) {
-			throw new IllegalArgumentException("Field 'item': required");
-		}
-		
-		logger.debug("Getting form for item: {}", itemId);
-		
+
+	public PublishRelease getPublishRelease(CmsItemIdArg itemId) throws Exception {
+
 		CmsItemLookupReporting cmsItemLookupReporting = lookup.get(itemId.getRepository());
 		CmsItem item = cmsItemLookupReporting.getItem(itemId);
 		CmsItemPublish itemPublish = new CmsItemPublish(item);
-		
 
 		PublishProfilingSet itemProfilingSet = publishConfiguration.getItemProfilingSet(itemPublish);
-		Map<String, PublishProfilingRecipe> itemProfilings = new HashMap<>(); //Initialize the map to prevent NPE in Velocity.
+		Map<String, PublishProfilingRecipe> itemProfilings = new HashMap<>(); // Initialize the map to prevent NPE in Velocity.
 		if (itemProfilingSet != null) {
 			itemProfilings = itemProfilingSet.getMap();
-			logger.debug("ItemId: {} has: {} configured profiles", itemId, itemProfilings.size());
 		}
-		
+
 		Map<String, PublishConfig> configuration = publishConfiguration.getConfigurationFiltered(itemPublish);
-		
-		// Avoid displaying an empty dialog, too complex to handle in Velocity (probably possible though).
-		if (configuration.isEmpty()) {
-			throw new IllegalStateException("No publications are configured.");
-		} else {
-			int visible = 0;
-			for (Entry<String, PublishConfig> config: configuration.entrySet()) {
-				if (config.getValue().isVisible()) {
-					visible++;
-				}
-			}
-			if (visible == 0) {
-				throw new IllegalStateException("No publications are configured to be visible.");
-			}
+
+		return new PublishRelease((CmsItemRepositem) item, configuration, itemProfilings);
+	}
+
+	@GET
+	@Path("release")
+	@Produces({MediaType.TEXT_HTML, MediaType.APPLICATION_JSON})
```

Comment:
Should HTML and JSON be the same gettor? The HTML does not need a full PublishRelease object?

Reply:
Ok, keep it. Required by objectwriter implementation.